sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -64,118 +71,60 @@ export interface LastEnqueuedEventProperties {
-   */
-  private readonly audience: string;
-  /**
-   * Provides relevant information about the amqp connection,
-   * cbs and $management sessions, token provider, sender and receivers.
-   */
-  private readonly _context: ConnectionContext;
-  /**
-   * The auth loop.
-   */
-  private authLoop?: TimerLoop;
-  /**
-   * The logger.
-   */
-  private readonly logger: SimpleLogger;
+  readonly isClosed: boolean;
+  readonly close: () => Promise<void>;
+  readonly abort: () => Promise<void>;
+  readonly isOpen: () => boolean;
+  readonly receiveBatch: (
+    maxMessageCount: number,
+    maxWaitTimeInSeconds?: number,
+    abortSignal?: AbortSignalLike
+  ) => Promise<ReceivedEventData[]>;
+  /** Needed for tests only */
+  readonly _onError?: (error: MessagingError | Error) => void;
+  readonly connect: (options: InitializeOptions) => Promise<void>;
+}
 
-  /**
-   * Instantiates a receiver that can be used to receive events over an AMQP receiver link in
-   * either batching or streaming mode.
-   * @param context -        The connection context corresponding to the EventHubClient instance
-   * @param consumerGroup -  The consumer group from which the receiver should receive events from.
-   * @param partitionId -    The Partition ID from which to receive.
-   * @param eventPosition -  The position in the stream from where to start receiving events.
-   * @param options -      Receiver options.
-   */
-  constructor(
-    context: ConnectionContext,
-    consumerGroup: string,
-    partitionId: string,
-    eventPosition: EventPosition,
-    options: EventHubConsumerOptions = {}
-  ) {
-    this.address = context.config.getReceiverAddress(partitionId, consumerGroup);
-    this.name = getRandomName(this.address);
-    this.audience = context.config.getReceiverAudience(partitionId, consumerGroup);
-    this._context = context;
-    this.ownerLevel = options.ownerLevel;
-    this.eventPosition = eventPosition;
-    this.options = options;
-    this.lastEnqueuedEventProperties = {};
-    const logPrefix = createLogPrefix(context.connectionId, "Receiver", this.name);
-    this.logger = createSimpleLogger(logger, logPrefix);
+interface InitializeOptions {
+  abortSignal: AbortSignalLike | undefined;
+  timeoutInMs: number;
+}
+
+/** @internal */
+export function createReceiver(
+  ctx: ConnectionContext,
+  consumerGroup: string,
+  partitionId: string,
+  eventPosition: EventPosition,
+  options: EventHubConsumerOptions = {}
+): PartitionReceiver {
+  const address = ctx.config.getReceiverAddress(partitionId, consumerGroup);
+  const name = getRandomName(address);
+  const audience = ctx.config.getReceiverAudience(partitionId, consumerGroup);
+  const ownerLevel = options.ownerLevel;
+  const logPrefix = createLogPrefix(ctx.connectionId, "Receiver", name);
+  const logger = createSimpleLogger(azureLogger, logPrefix);
+
+  let receiver: Receiver | undefined = undefined;
+  let authLoop: TimerLoop;
+  let isConnecting = false;
```

Comment:
These are the only things that are part of the state that is mutable and the reason being that the connection is not established at the time of the construction. I will investigate in another PR if we can connect at the construction time to make the first two constants as well. If it turns out we can do that, it will simplify the receiver object being returned as the `receiveBatch` operation won't have to worry about it.

Reply:
It looks like the underlying link could be closed but the receiver is still up so the `receiveBatch` will always have to handle connection in case the link was closed.